go/vt/vtgate/planbuilder/abstract/operator.go

```diff
@@ -287,6 +289,56 @@ func createOperatorFromUpdate(updStmt *sqlparser.Update, semTable *semantics.Sem
 	return subq, nil
 }
 
+func createOperatorFromDelete(deleteStmt *sqlparser.Delete, semTable *semantics.SemTable) (LogicalOperator, error) {
+	subq, err := createSubqueryFromStatement(deleteStmt, semTable)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
nit: we could move this to line 335 where we do a nil check (`subq == nil`), so we don't have to prepare the subquery if the delete statement has to fail

Reply:
Fixed